### PR TITLE
Added user-defined log level functionality to hapsequencer.py, runsinglehap.py, ci_table.py, config_utils.py, and hla_flag_filter.py

### PR DESCRIPTION
**Pull request by dulude**
- Added user-defined log level functionality to the following:
     - drizzlepac/hapsequencer.py
     - drizzlepac/runsinglehap.py
     - drizzlepac/hlautils/ci_table.py
     - drizzlepac/hlautils/config_utils.py
     - drizzlepac/hlautils/hla_flag_filter.py
- Streamlined logging level setup in runsinglehap.perform()
- drizzlepac/hapsequencer.py: updated formatting of log messages written to log file to match the formatting of log messages displayed on-screen

- See issue #502.
- The code being merged into master in PR #518 was originally branched from commit [ca46a9b](https://github.com/spacetelescope/drizzlepac/pull/510/commits/ca46a9bae2a7ce25f8b25885f78bfb39442d6852) from this PR and thus includes all of changes and updates in that commit.

### CODE REVIEW

File: drizzlepac/hlautils/hla_flag_filter.py [hunk 1]
```diff
@@ -641,6 +657,7 @@ def hla_swarm_flags(drizzled_image, catalog_name, catalog_data, exptime, plate_s
 
     ap2 = param_dict['catalog generation']['aperture_2']
     if proc_type not in ('segment', 'aperture'):
+        log.error("Unknown proc_type '{}', must be 'aperture' or 'segment'".format(proc_type))
```
Comment (stsci-hack) [inline drizzlepac/hlautils/hla_flag_filter.py]:
```suggestion
        log.error("Unknown catalog type '{}', must be 'aperture' or 'segment'".format(proc_type))
```
This is a little more understandable to an end user, rather referring to an internal variable.

File: drizzlepac/hlautils/hla_flag_filter.py [hunk 2]
```diff
@@ -1438,8 +1455,10 @@ def xymatch(cat1, cat2, sep, multiple=False, stack=True, verbose=True):
     Varies; Depending on inputs, either just 'p2', or 'p1' and 'p2'. p1 and p2 are lists of matched indices
     """
     if not (isinstance(cat1, numpy.ndarray) and len(cat1.shape) == 2 and cat1.shape[1] == 2):
+        log.error("cat1 must be a [N, 2] array")
```
Comment (stsci-hack) [inline drizzlepac/hlautils/hla_flag_filter.py]:
```suggestion
        log.error("catalog 1 must be a [N, 2] array")
```

File: drizzlepac/hlautils/hla_flag_filter.py [hunk 3]
```diff
@@ -1438,8 +1455,10 @@ def xymatch(cat1, cat2, sep, multiple=False, stack=True, verbose=True):
...
         raise ValueError("cat1 must be a [N, 2] array")
     if not (isinstance(cat2, numpy.ndarray) and len(cat2.shape) == 2 and cat2.shape[1] == 2):
+        log.error("cat2 must be a [N, 2] array")
```
Comment (stsci-hack) [inline drizzlepac/hlautils/hla_flag_filter.py]:
```suggestion
        log.error("catalog 2 must be a [N, 2] array")
```

**Comment by dulude** [comment]
> Please check code against PEP8 requirements and update as necessary. The "long line" complaint can be ignored as appropriate.

Hey @mdlpstsci, I fixed the PEP8 issues. Commit [1cc6f5b](https://github.com/spacetelescope/drizzlepac/pull/510/commits/1cc6f5b8670b0514e49e2ccc0650ed81450f8833) has all pep8 issues resolved.

**Pull request merged**